src/metatags/seo-checks.js

```diff
@@ -0,0 +1,209 @@
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  DESCRIPTION, TITLE, H1, TAG_LENGTHS, ISSUE, ISSUE_DETAILS, SEO_IMPACT, HIGH,
+  SEO_RECOMMENDATION, SHOULD_BE_PRESENT, TITLE_LENGTH_SUGGESTION,
+  DESCRIPTION_LENGTH_SUGGESTION, H1_LENGTH_SUGGESTION, MODERATE,
+  ONE_H1_ON_A_PAGE, UNIQUE_ACROSS_PAGES, DUPLICATES, MULTIPLE_H1_ON_PAGE,
+} from './constants.js';
+
+class SeoChecks {
+  constructor(log) {
+    this.log = log;
+    this.detectedTags = {};
+    this.allTags = {
+      [TITLE]: {},
+      [DESCRIPTION]: {},
+      [H1]: {},
+    };
+  }
+
+  /**
+   * Capitalises the first character of a given string
+   * @param str
+   * @returns {string}
+   */
+  static #capitalizeFirstLetter(str) {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  }
+
+  /**
+   * Checks for missing tags on the page and adds to detected tags array if found lacking.
+   * @param {string} url - The URL of the page.
+   * @param {object} pageTags - An object containing the tags of the page.
+   */
+  checkForMissingTags(url, pageTags) {
+    [TITLE, DESCRIPTION, H1].forEach((tagName) => {
+      if (pageTags[tagName] === undefined
+        || (Array.isArray(pageTags[tagName]) && pageTags[tagName].length === 0)) {
+        const capitalisedTagName = SeoChecks.#capitalizeFirstLetter(tagName);
+        this.detectedTags[url] ??= {};
+        this.detectedTags[url][tagName] = {
+          [SEO_IMPACT]: HIGH,
+          [ISSUE]: `Missing ${capitalisedTagName}`,
+          [ISSUE_DETAILS]: `${capitalisedTagName} tag is missing`,
+          [SEO_RECOMMENDATION]: SHOULD_BE_PRESENT,
+        };
+      }
+    });
+  }
+
+  /**
+   * Checks if tag lengths are within recommended limits
+   * and adds to detected tags array if found lacking.
+   * @param {string} url - The URL of the page.
+   * @param {object} pageTags - An object containing the tags of the page.
+   */
+  checkForTagsLength(url, pageTags) {
+    const getLengthSuggestion = (tagName) => {
+      if (TITLE === tagName.toLowerCase()) {
+        return TITLE_LENGTH_SUGGESTION;
+      } else if (DESCRIPTION === tagName.toLowerCase()) {
+        return DESCRIPTION_LENGTH_SUGGESTION;
+      }
+      return H1_LENGTH_SUGGESTION;
+    };
+
+    const checkTag = (tagName, tagContent) => {
+      const capitalizedTagName = SeoChecks.#capitalizeFirstLetter(tagName);
+      let issueDetails;
+      let issueImpact;
+      let issue;
+      let recommendation;
+
+      if (tagContent === '') {
```

Comment:
use `hasText`

Reply:
Here I would like to specifically identify if the string is empty. The hasText method will return false for null, undefined, number, empty string etc.